api/v1beta1/common_types.go

```diff
@@ -64,6 +64,11 @@ type WatcherCommon struct {
 	// to /etc/<service>/<service>.conf.d directory as a custom config file.
 	CustomServiceConfig string `json:"customServiceConfig,omitempty"`
 
+	// +kubebuilder:validation:Required
+	// +kubebuilder:default=metric-storage-prometheus-config
```

Comment:
required filed should not have a default.
if they have a default the are optional.

operator lint is meant to catch this but i don't actual see a lint rule in place for this edgecase.

https://github.com/gibizer/operator-lint/tree/main/linters/crd

Reply:
That name was proposed by the CloudOps team as the default secret name that will be created by the telemetry-operator with the required info.

I'm doing it optional, thanks!